File: BeerPlease/app/src/main/java/app/dgandroid/eu/beerplease/utils/Logger.java
package app.dgandroid.eu.beerplease.utils;

import android.util.Log;

/**
 * Created by Duilio on 22/05/2017.
 */

public class Logger {
    public static void i(String msg) {
        if (Config.DEBUG){
            Log.println(Log.INFO, getTag(), msg);
        }
    }
    public static void e(String msg) {
        if (Config.DEBUG){
            Log.println(Log.ERROR, getTag(), msg);
        }
    }
    private static String getTag() {
        String className = new Exception().getStackTrace()[2].getClassName();
        return className.substring(1 + className.lastIndexOf('.'));
    }
}

File: BeerPlease/app/src/main/java/app/dgandroid/eu/beerplease/utils/Logger.kt
package app.dgandroid.eu.beerplease.utils

import android.util.Log

/**
 * Created by Duilio on 22/05/2017.
 */

object Logger {
    private val tag: String
        get() {
            val className = Exception().stackTrace[2].className
            return className.substring(1 + className.lastIndexOf('.'))
        }

    fun i(msg: String) {
        if (Config.DEBUG) {
            Log.println(Log.INFO, tag, msg)
        }
    }

    fun e(msg: String) {
        if (Config.DEBUG) {
            Log.println(Log.ERROR, tag, msg)
        }
    }
}